Jenkinsfile

```diff
@@ -1,90 +1,55 @@
 #!/usr/bin/env groovy
 
 pipeline {
-  agent any
+  agent none
   environment {
-    HOME = "${env.HUDSON_HOME}"
     BASE_DIR="src/go.elastic.co/apm"
-    JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
   }
   options {
     timeout(time: 1, unit: 'HOURS')
-    buildDiscarder(logRotator(numToKeepStr: '3', artifactNumToKeepStr: '2', daysToKeepStr: '30'))
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
     timestamps()
-    preserveStashes()
     ansiColor('xterm')
     disableResume()
     durabilityHint('PERFORMANCE_OPTIMIZED')
   }
   parameters {
-    string(name: 'branch_specifier', defaultValue: "", description: "the Git branch specifier to build (<branchName>, <tagName>, <commitId>, etc.)")
     string(name: 'GO_VERSION', defaultValue: "1.10.3", description: "Go version to use.")
     booleanParam(name: 'Run_As_Master_Branch', defaultValue: false, description: 'Allow to run any steps on a PR, some steps normally only run on master branch.')
-    booleanParam(name: 'linux_ci', defaultValue: true, description: 'Enable Linux build')
     booleanParam(name: 'test_ci', defaultValue: true, description: 'Enable test')
-    booleanParam(name: 'integration_test_ci', defaultValue: true, description: 'Enable run integration test')
-    booleanParam(name: 'integration_test_pr_ci', defaultValue: false, description: 'Enable run integration test')
-    booleanParam(name: 'integration_test_master_ci', defaultValue: false, description: 'Enable run integration test')
+    booleanParam(name: 'docker_test_ci', defaultValue: true, description: 'Enable run docker tests')
     booleanParam(name: 'bench_ci', defaultValue: true, description: 'Enable benchmarks')
     booleanParam(name: 'doc_ci', defaultValue: true, description: 'Enable build documentation')
   }
   stages {
-    /**
-     Checkout the code and stash it, to use it on other stages.
-    */
-    stage('Checkout') {
+    stage('Initializing'){
```

Comment:
this stage run checkout and build on the same agent, avoid to provision a new agent for the default build.